main.tf

```diff
@@ -3006,6 +3006,21 @@ resource "aws_iam_instance_profile" "karpenter" {
   tags = merge(var.tags, try(var.karpenter_node.instance_profile_tags, {}))
 }
 
+resource "aws_eks_access_entry" "node" {
+  count = var.enable_karpenter && var.karpenter_create_access_entry ? 1 : 0
+
+  cluster_name  = var.cluster_name
+  principal_arn = local.create_karpenter_node_iam_role ? aws_iam_role.karpenter[0].arn : var.karpenter.node_iam_role_arn
+  type          = "EC2_LINUX"
+
+  tags = var.tags
+
+  depends_on = [
+    # If we try to add this too quickly, it fails. So .... we wait
+    module.karpenter_sqs
```

Comment:
Will it work if karpenter_enable_spot_termination = false? 
Because sqs were not created in this case.